service/src/main/java/uk/nhs/adaptors/gp2gp/tasks/TaskExecutorFactory.java

```diff
@@ -0,0 +1,23 @@
+package uk.nhs.adaptors.gp2gp.tasks;
+
+import java.util.Map;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import lombok.RequiredArgsConstructor;
+
+@Component
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class TaskExecutorFactory {
+    private static final Map<String, TaskExecutor> TASK_EXECUTOR_MAP;
+
+    static {
```

Comment:
I am not sure of this is necessary. Did you try assigning this map directly to TASK_EXECUTOR_MAP?